xcdat/averager_accessor.py

```diff
@@ -0,0 +1,506 @@
+        lon_bounds : Optional[RegionAxisBounds], optional
+            Tuple of longitude boundaries for regional selection, by default None.
+        is_weighted : bool, True
+            Weighting option to calculate weights proportional to area
+            (True), uniform weights (False).
+
+        Returns
+        -------
+        AxisWeights
+            Set of weights for each eligible dimension.
+        """
+        if is_weighted:
+            weights = self._get_area_weights(lat_bounds, lon_bounds)
+        else:
+            weights = self._get_uniform_weights(xvariable)
+        return weights
+
+    def _get_area_weights(
+        self,
+        lat_bounds: Optional[RegionAxisBounds],
+        lon_bounds: Optional[RegionAxisBounds],
+    ) -> AxisWeights:
+        """Gets area weights for a dimension with weighted bounds.
+
+        # TODO: Add more description on how weights are generated.
+
+        Parameters
+        ----------
+        lat_bounds : Optional[RegionAxisBounds], optional
+            Tuple of latitude boundaries for regional selection, by default None.
+        lon_bounds : Optional[RegionAxisBounds], optional
+            Tuple of longitude boundaries for regional selection, by default None.
+
+        Returns
+        -------
+        AxisWeights
+            Set of weights for each eligible dimension.
+
+        Notes
+        -----
+        This is a prototype and needs work to make it work more generically. It
+        currently only works with lat / lon. It might be useful to check the
+        latitude axis units.
+        """
+        BoundsByType = TypedDict(
+            "BoundsByType",
+            {"domain": Optional[xr.DataArray], "region": Optional[RegionAxisBounds]},
+        )
+        bounds: Dict[str, BoundsByType] = {
+            "lat": {
+                "domain": self._dataset.bounds.get_bounds("lat"),
+                "region": lat_bounds,
+            },
+            "lon": {
+                "domain": self._dataset.bounds.get_bounds("lon"),
+                "region": lon_bounds,
+            },
+        }
+        weights: AxisWeights = {}
+
+        for dim, bounds_by_type in bounds.items():
+            domain_bounds = bounds_by_type["domain"]
+            region_bounds = bounds_by_type["region"]
+
+            if domain_bounds is not None:
+                domain_bounds_copy = domain_bounds.copy()
+
+                if region_bounds is not None:
+                    # ensure region bounds can be indexed and are type(float)
+                    region_bounds_arr = np.array(region_bounds).astype(float)
+
+                    # TODO: Refactor this chunk of code into a private function
+                    # =========================================================
+                    if dim == "lon":
+                        # swap region + dim bounds to be 360-type
+                        domain_bounds[:] = self._swap_region_lon_axis(
```

Comment:
@tomvothecoder - I think this line is problematic in some situations. Should `domain_bounds` be `domain_bounds_copy` (here and below)?

Reply:
Good catch, I just pushed a fix